cpp/tensorrt_llm/thop/loraOp.cpp

```diff
@@ -172,6 +175,197 @@ std::vector<th::Tensor> lora_grouped_gemm(th::Tensor const& input, th::Tensor co
+
+    auto* host_max_in_sizes_ptr = reinterpret_cast<cutlass::gemm::GemmCoord*>(host_max_in_sizes.data_ptr());
+    auto* host_max_out_sizes_ptr = reinterpret_cast<cutlass::gemm::GemmCoord*>(host_max_out_sizes.data_ptr());
+
+    // Step 4: Get weight pointers using CUDA Graph compatible operations
+    // The b_ptrs and b_prime_ptrs tensors already contain pointer values as int64
+    // Cast them directly to void* arrays for the GEMM kernels (remove const for CUTLASS compatibility)
+
+    auto* b_ptrs_gpu = reinterpret_cast<void**>(const_cast<void*>(b_ptrs.data_ptr()));
+    auto* b_prime_ptrs_gpu = reinterpret_cast<void**>(const_cast<void*>(b_prime_ptrs.data_ptr()));
+
+    // Step 4.5: Get precomputed leading dimension pointers for GEMM operations (remove const for CUTLASS compatibility)
+    auto* lda_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(lda.data_ptr()));
+    auto* ldb_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(ldb.data_ptr()));
+    auto* ldd_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(ldd.data_ptr()));
+    auto* ldb_prime_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(ldb_prime.data_ptr()));
+    auto* ldd_prime_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(ldd_prime.data_ptr()));
+
+    auto* splitk_offsets_gpu = reinterpret_cast<int64_t*>(const_cast<void*>(splitk_offsets.data_ptr()));
+
+    // Step 5: Direct CUTLASS grouped GEMM setup (no CuBLAS wrapper needed)
+    // The new CUDA Graph compatible implementation uses CUTLASS directly
+
+    // Get data type
+    nvinfer1::DataType loraRuntimeDataType;
+    switch (dtype)
+    {
+    case torch::kFloat16: loraRuntimeDataType = nvinfer1::DataType::kHALF; break;
+    case torch::kBFloat16: loraRuntimeDataType = nvinfer1::DataType::kBF16; break;
+    default: TORCH_CHECK(false, "Invalid dtype, only supports float16, bfloat16, got %s", c10::toString(dtype));
+    }
+
+    int const minKnInt = std::max(1, static_cast<int>(minKN));
+
+    // Step 6: Call CUDA Graph compatible grouped GEMM for lora_in (split-K)
+    // No parameter workspace needed - tensors are already on GPU
+    if (problem_count > 0)
+    {
+        TLLM_LOG_TRACE("Start Grouped GEMM for LoRA in.");
+
+        tk::cuda_graph_splitk_grouped_gemm(problem_sizes_1_ptr, problem_count, a_ptrs_gpu, b_ptrs_gpu,
+            d_ptrs_gpu,                                     // ptrC (no bias)
+            d_ptrs_gpu, lda_gpu, ldb_gpu, ldd_gpu, ldd_gpu, // Precomputed leading dimensions
+            true,                                           // isLoraIn
+            loraRuntimeDataType,
+            16,                                             // splitKSlices
+            minKnInt,                                       // minKN
+            host_max_in_sizes_ptr, splitk_offsets_gpu, stream);
+        sync_check_cuda_error(stream);
+    }
+
+    // Step 7: Call CUDA Graph compatible grouped GEMM for lora_out
+    if (problem_count > 0)
+    {
```

Comment:
```suggestion

        // Step 7: Call CUDA Graph compatible grouped GEMM for lora_out
```
If accepted, I would also move the `// Step 6:` comment to be inside the `if`